src/index.tsx

```diff
@@ -51,17 +53,26 @@ export interface PlayerRef {
 }
 
 export const Player = forwardRef<PlayerRef, Props>(function Player(
-    props: Props,
+    {
+        events,
+        onPlayerTimeChange,
+        onPrevious,
+        onNext,
+        isBuffering = false,
+        duration = 0
+    }: Props,
     ref: ForwardedRef<PlayerRef>
 ) {
     const [playing, setPlaying] = useState(true)
     const [skipping, setSkipping] = useState(false)
     const [currentTime, setCurrentTime] = useState(0)
-    const [meta, setMeta] = useState<playerMetaData>({
-        startTime: 0,
-        endTime: 0,
-        totalTime: 0
-    })
+    const durationMs = convertSecondsToMillis(duration)
```

Comment:
Why not send the duration as milliseconds from the backend? This avoids back-and-forth conversions.